### ISSUE TITLE
Can't run nmf-test-opencl on gallium, opencl-mesa

### ISSUE THREAD

**Issue by nartes** [viennacl/viennacl-dev#229]
https://gist.github.com/anonymous/4e947b651674b6b26c3972651b5ef71e

I've tried to compile the upstream code, but few errors appeared related to exp10 is not present in std
namespace, yet another function had the same problem.
Then I switched to the tag 1.7.0. It has been built successfully.
But when I tried to run nmf-test-opencl, a runtime errors occured in gallium library,
provided by opencl-mesa.

Perhaps, I should try with proprietary opencl-catalyst driver based on fglrx kernel module.

**Comment by karlrupp** [comment]
Yes, please try the proprietary catalyst driver. The code fails in the matrix-matrix products, for which we have a built-in database of different device parameters. Chances are that the opencl-mesa driver just chokes... Please let me know whether the driver change fixes the problem.
